Assets/HoloToolkit/UX/Scripts/BoundingBoxes/BoundingBox.cs

```diff
@@ -0,0 +1,482 @@
+            }
+            protected set
+            {
+                if (flattenedAxis != value)
+                {
+                    flattenedAxis = value;
+                    if (OnFlattenedAxisChange != null)
+                        OnFlattenedAxisChange();
+                }
+            }
+        }
+
+        #endregion
+
+        #region private
+
+        /// <summary>
+        /// Override so we're not overwhelmed by button gizmos
+        /// </summary>
+        #if UNITY_EDITOR
+        protected virtual void OnDrawGizmos()
+        {
+            // nothing
+            if (!Application.isPlaying)
+            {
+                // Do this here to ensure continuous updates in editor
+                CreateTransforms();
+                RefreshTargetBounds();
+                UpdateScaleTransform();
+            }
+
+            if (target != null)
+            {
+                foreach (Vector3 point in boundsPoints)
+                {
+                    Gizmos.DrawSphere(target.transform.TransformPoint(point), 0.01f);
+                }
+            }
+        }
+        #endif
+
+        protected virtual void Update()
+        {
+            CreateTransforms();
+            RefreshTargetBounds();
+            UpdateScaleTransform();
+        }
+
+        protected virtual void CreateTransforms()
+        {
+            if (scaleTransform == null)
+                scaleTransform = transform.Find("Scale");
+
+            if (scaleTransform == null)
+                scaleTransform = new GameObject("Scale").transform;
+
+            scaleTransform.parent = transform;
+        }
+
+        protected virtual void RefreshTargetBounds()
+        {
+            if (target == null)
+            {
+                targetBoundsWorldCenter = Vector3.zero;
+                targetBoundsLocalScale = Vector3.one;
+                return;
+            }
+
+            // Get the new target bounds
+            boundsPoints.Clear();
+
+            switch (boundsCalculationMethod)
+            {
+                case BoundsCalculationMethodEnum.RendererBounds:
+                default:
+                    Renderer[] renderers = target.GetComponentsInChildren<Renderer>();
+                    for (int i = 0; i < renderers.Length; ++i)
+                    {
+                        var rendererObj = renderers[i];
+                        if (rendererObj.gameObject.layer == ignoreLayer)
+                            continue;
+
+                        rendererObj.bounds.GetCornerPositionsFromRendererBounds(ref corners);
+                        boundsPoints.AddRange(corners);
+                    }
+                    break;
+
+                case BoundsCalculationMethodEnum.Colliders:
+                    Collider[] colliders = target.GetComponentsInChildren<Collider>();
+                    for (int i = 0; i < colliders.Length; i++)
+                    {
+                        switch (colliders[i].GetType().Name)
```

Comment:
Let's break these nested switch statements into their own methods so they're easier to follow.